_posts/2017-07-06-hitchhikers-guide-to-cp.md

```diff
@@ -0,0 +1,308 @@
+
+Now the real question, **"Why should one try it?"**. Well, the answer is simple, programming is a major portion of the work that is done in tech companies. Being good at programming makes it easier to get a job in a good tech company. Competitive programming is the best way of getting good at programming and problem solving. Apart from this, one gets recognition and meets other programmers. There are prizes for top performers in each of these competitions. Most importantly, it quenches the thirst for those looking for challenges.
+
+The very first thing one should note is that this guide is only meant to help you get started with competitive programming and not meant to explain concepts in detail. Anyone interested in further reading can refer to the resources provided at the very end of this guide.
+
+---
+
+# Prerequisite
+
+Basic knowledge of any programming language. Generally, C, C++, JAVA and PYTHON are used for CP. We'll use C++. The reason behind using C++ and not C is that C++ contains templates of certain frequently used Algorithms and Data Strutures, commonly known as STL. Using them makes the code smaller and easier to debug. Also, PYTHON is helpful in problems with big integers.
+
+##  Some Language tutorials:
+
++   **PYTHON :** [HackerEarth](https://www.hackerearth.com/practice/python/), [Python Documentation](https://docs.python.org/3/tutorial/index.html)
+
++   **STL :** [HackerEarth](https://www.hackerearth.com/practice/notes/standard-template-library/)
+
++   **C Video Lectures :** [CS50](https://www.youtube.com/watch?v=o4SGkB_8fFs&list=PLhQjrBD2T382VRUw5ZpSxQSFrxMOdFObl) (Also gets you familiar with other fundamentals of Comp. Science).
+
+---
+
+# Online Platforms
+
+## Practice:
+
++   **[SPOJ](https://www.spoj.com) :** 
+
+    +   Contains a lot of programming problems of varied difficulty.
+
+    +   Start with problems having maximum submissions and move on to ones with lesser submissions once you build some confidence.  
+
+    +   Start participating in Codechef and Codeforces contests once you have solved at least **30-40** problems.
+
++   **[CodeMonk](https://www.hackerearth.com/practice/codemonk/) :**
+
+    +   Contains both tutorials and problems of basic Algorithms and Data Structures.
+
+    +   Tutorials are easy to understand and covers most of the topic.
+
+    +   Can be easily completed within a month or two.
+
+    +   A **'must do'** for beginners.
+
++   **[A2OJ](https://www.a2oj.com) :**
+
+    +   We are often unable to find good problems of a particular topic, that's where a2oj comes to rescue.
+
+    +   Problems are from online judges like SPOJ, UVa, Codeforces and are classified according to topic and difficulty.
+
+## Contests:
```

Comment:
I wanted to add heuristics, but then I thought that hackerEarth isn't that good a judge as Codechef and Codeforces. But, I'll add that without adding hackerEarth :P.